formats/img-gif.js

```diff
@@ -0,0 +1,673 @@
+						reason: `File contains an extension, which is not supported.`,
+					};
+				}
+
+			} else if (introducer === IMG_DESC_INTRODUCER) {
+
+				if (buffer.distFromEnd() >= 9) {
+
+					const imgDesc = buffer.readRecord(recordTypes.imageDescriptor);
+
+					// If a local color table is being used, calculate its size so that
+					// we can jump past it in the buffer.
+					let localColorTableSize = 0;
+					if (imgDesc.packedField & 0x80) {
+						localColorTableSize = (1 << ((imgDesc.packedField & 0x07) + 1)) * 3;
+					}
+
+					if (buffer.distFromEnd() >= localColorTableSize) {
+						buffer.seekRel(localColorTableSize);
+					} else {
+						return {
+							valid: false,
+							reason: `File contains an incomplete local color table.`,
+						};
+					}
+
+					// Need at least two more bytes to get started reading the image data.
+					if (buffer.distFromEnd() >= 2) {
+
+						// Check that the LZW minimum code size is valid
+						const lzwMinCodeSize = buffer.read(RecordType.int.u8);
+						if ((lzwMinCodeSize < 2) || (lzwMinCodeSize > 12)) {
+							return {
+								valid: false,
+								reason: `LZW minimum code size (${lzwMinCodeSize}) is outside valid range for GIF (2 to 12).`,
+							};
+						}
+
+						let subBlockLen = buffer.read(RecordType.int.u8);
+						while ((subBlockLen > 0) && (buffer.distFromEnd() > 0)) {
+							// Skip past this subblock
+							buffer.seekRel(subBlockLen);
+							subBlockLen = (buffer.distFromEnd() > 0) ? buffer.read(RecordType.int.u8) : 0;
+						}
+
+						if (subBlockLen == 0) {
+							foundImageData = true;
+						} else {
+							return {
+								valid: false,
+								reason: `File contains an incomplete image data subblock.`,
+							};
+						}
+					} else {
+						return {
+							valid: false,
+							reason: `File contains an incomplete image data block.`,
+						};
+					}
+				} else {
+					return {
+						valid: false,
+						reason: `File contains insufficient data to complete image descriptor.`,
+					};
+				}
+			} else if (introducer == TRAILER) {
+				foundTrailer = true;
+			}
+		}
+
+		if (!foundImageData) {
+			return {
+				valid: false,
+				reason: `File ends without containing at least one image data block.`,
+			};
+		}
+
+		return {
+			valid: true,
+			reason: `Potentially valid GIF image, pending decode.`,
+		};
+	}
+
+	static read(content) {
+
+		const extensionsAllowed = this.getExtensionsAllowed();
+		let buffer = new RecordBuffer(content.main);
+		let gifFrames = [];
+
+		// We've previously identified this file, so we can skip past the header.
+		buffer.seekRel(HEADER_MAGIC_LEN);
+
+		const logicalDesc = buffer.readRecord(recordTypes.logicalScreenDescriptor);
+		const usingGlobalColor = logicalDesc.packedField & 0x80;
+		const globalColorBpp = (logicalDesc.packedField & 0x07) + 1;
+		const globalTableColorCount = (1 << globalColorBpp);
+
+		let globalPal = new Palette(globalTableColorCount);
+
+		// If a local color table is being used...
```

Comment:
Is this meant to say 'global' rather than 'local'?

Reply:
Fixed in latest commit.